### ISSUE TITLE
`tf.nn.conv2d_transpose` crashes with "Illegal instruction (core dumped)"

### ISSUE THREAD

**Issue by jiren-the-gray** [tensorflow/tensorflow#93733]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

binary

### TensorFlow version

tf 2.19

### Custom code

Yes

### OS platform and distribution

Ubuntu 24.04.2 LTS

### Python version

3.12

### CUDA/cuDNN version

12.6

### Current behavior?

When I run the following code setting the device as CPU with `TF_ENABLE_ONEDNN_OPTS=1`, it crashes with this error message: "could not create a primitive descriptor for a convolution forward propagation primitive", then it raises the signal "Illegal instruction (core dumped)". I tried it with two different machines, both gave me the same error.

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import numpy as np
rng = np.random.default_rng(40)

tf.config.experimental.enable_op_determinism()

with tf.device("/CPU:0"):
    input_tensor = tf.constant(rng.uniform(-3.9808033, -3.0141976, (1, 4, 4, 5)), dtype=tf.float32)
    weight_tensor = tf.constant(rng.uniform(2.0, 2.0, (4, 4, 4, 4)), dtype=tf.float32)
    stride = 2
    padding = 3
    output_padding = 0
    groups = 1
    dilation = 1
    
    input_tensor = tf.transpose(input_tensor, [0, 2, 3, 1])
    weight_tensor = tf.transpose(weight_tensor, [2, 3, 1, 0])

    batch_size = tf.shape(input_tensor)[0]
    input_height = input_tensor.shape[1]
    input_width = input_tensor.shape[2]
    filter_height = weight_tensor.shape[0]
    filter_width = weight_tensor.shape[1]
    
    out_channels = weight_tensor.shape[3]
    
    h_out = (input_height - 1) * stride - 2 * padding + dilation * (filter_height - 1) + 1 + output_padding
    w_out = (input_width - 1) * stride - 2 * padding + dilation * (filter_width - 1) + 1 + output_padding
    output_shape = [batch_size, h_out, w_out, out_channels]

    output = tf.nn.conv2d_transpose(
        input_tensor, weight_tensor, output_shape=output_shape, strides=[1, stride, stride, 1],
        padding='VALID' if padding == 0 else 'SAME', dilations=[1, dilation, dilation, 1]
    )
```

### Relevant log output

```shell
could not create a primitive descriptor for a convolution forward propagation primitiveIllegal instruction (core dumped)
```

**Comment by jiren-the-gray** [comment]
Hi @Venkat6871 , thanks for the reply. I can see that the error is not reproducible on colab but I have tried running this on two different machines and it did crash on them, even after copying the code from your gist, it crashed. Here is the list of dependencies in the environment where I am running this code:
```
Package                 Version
----------------------- ---------
absl-py                 2.2.2
astunparse              1.6.3
certifi                 2025.4.26
charset-normalizer      3.4.2
flatbuffers             25.2.10
gast                    0.6.0
google-pasta            0.2.0
grpcio                  1.71.0
h5py                    3.13.0
idna                    3.10
keras                   3.10.0
libclang                18.1.1
Markdown                3.8
markdown-it-py          3.0.0
MarkupSafe              3.0.2
mdurl                   0.1.2
ml_dtypes               0.5.1
namex                   0.0.9
numpy                   2.1.3
opt_einsum              3.4.0
optree                  0.15.0
packaging               25.0
pip                     25.1
protobuf                5.29.4
Pygments                2.19.1
requests                2.32.3
rich                    14.0.0
setuptools              78.1.1
six                     1.17.0
tensorboard             2.19.0
tensorboard-data-server 0.7.2
tensorflow              2.19.0
termcolor               3.1.0
typing_extensions       4.13.2
urllib3                 2.4.0
Werkzeug                3.1.3
wheel                   0.45.1
wrapt                   1.17.2

```

**Comment by jiren-the-gray** [comment]
@Venkat6871 Hi, I was able to reproduce the issue on google colab as well by setting `TF_ENABLE_ONEDNN_OPTS = 1`. Here is the [gist](https://colab.research.google.com/drive/1HjnbDxltuLf5v7zAY-i_due413oIIc9i?usp=sharing)

**Comment by gzmkl** [comment]
It seems that the inputs are not valid.  I tested on a SkyLake machine with and without ONEDNN enabled. Here are error messages.
1.  TF_ENABLE_ONEDNN_OPTS=0
     "Local rendezvous is aborting with status: INVALID_ARGUMENT: Conv2DCustomBackpropInput: Size of out_backprop doesn't match computed: actual = 4, computed = 2 spatial_dim: 1 input: 4 filter: 4 output: 4 stride: 2 dilation: 1 ... "
2. TF_ENABLE_ONEDNN_OPTS=1
     … Operation received an exception:Status: 2, message: could not create a primitive descriptor for the convolution forward propagation primitive. Run workload with environment variable ONEDNN_VERBOSE=all to get additional diagnostic information ..."

**Comment by Venkat6871** [comment]
Hi @gzmkl, thanks for verifying on your side.
Hi @jiren-the-gray, apologies for the delay and thank you for your patience.
I tested your code on a VM running `Ubuntu 24.10`. On my setup, the code does not crash but instead raises a clear error message. I made some adjustments (mainly around padding), and with those changes, it now runs successfully.

I have attached a couple of screenshots showing the results on my machine for your reference.

<img width="1728" height="364" alt="Image" src="https://github.com/user-attachments/assets/1edf945b-5d92-445d-95e0-6ee97a2e247e" />

, 

<img width="1728" height="1117" alt="Image" src="https://github.com/user-attachments/assets/76a1dcf9-b0f4-4dc7-a1f8-fe62c06b72f1" />

Thank you!

**Comment by jiren-the-gray** [comment]
@Venkat6871 I understand that without setting `TF_ENABLE_ONEDNN_OPTS=1` it does not crash (may be due to being an invalid input) , and with some modification the error goes away completely. However, with ONEDNN enabled in multiple machines I have tested as well as colab, it is crashing. Reproducing is somewhat flaky however, for example if the code is run with `TF_ENABLE_ONEDNN_OPTS=0` once and then run with `TF_ENABLE_ONEDNN_OPTS=1` it sometimes does not crash, but the crash is still happening from time to time. Shouldn't the validity check close gracefully every time even on invalid inputs?
